org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/DeviceRequestTest.java

```diff
@@ -240,14 +281,55 @@ public void testMandelModelConfigure() throws Exception {
 		model.setEscapeRadius(15);
 		req.setDeviceModel(model);
 		req.setDeviceAction(DeviceAction.CONFIGURE);
-		
+
 		res = requester.post(req);
-        
-		info = (DeviceInformation<MandelbrotModel>)res.getDeviceInformation();
-		if (info==null) throw new Exception("There were no devices found and at least the mandelbrot example should have been!");
-		assertTrue(model.getExposureTime()==100); // We do not set an exposure as part of the test.
-		assertTrue(model.getEscapeRadius()==15); // We do not set an exposure as part of the test.
-		assertTrue(info.getState()==DeviceState.ARMED); // We do not set an exposure as part of the test.
+
+		@SuppressWarnings("unchecked")
+		DeviceInformation<MandelbrotModel> info2 = (DeviceInformation<MandelbrotModel>)res.getDeviceInformation();
+		assertNotNull("There were no devices found and at least the mandelbrot example should have been!", info2);
+		assertEquals(100, model.getExposureTime(), 1e-15); // We do not set an exposure as part of the test.
+		assertEquals(15, model.getEscapeRadius(), 1e-15); // We do not set an exposure as part of the test.
+		assertEquals(DeviceState.ARMED, info2.getState()); // We do not set an exposure as part of the test.
+	}
+
+	@Test
+	public void testGetAttribute() throws Exception {
+		DeviceRequest req = new DeviceRequest();
+		req.setDeviceName("malcolm");
+		req.setAttributeName("axesToMove");
+		DeviceRequest res = requester.post(req);
+		assertNotNull(res);
+		assertNotNull(res.getAttributes());
+		assertEquals(1, res.size());
+		IDeviceAttribute<?> attr = res.getAttributes().get("axesToMove");
+		assertNotNull(attr);
+		assertEquals("axesToMove", attr.getName());
+		assertEquals("axesToMove", attr.getLabel());
+		assertEquals("Default axis names to scan for configure()", attr.getDescription());
+		assertEquals(StringArrayAttribute.class, attr.getClass());
+		assertArrayEquals(new String[] { "stage_x", "stage_y" }, (String[]) attr.getValue());
+	}
+
+	@Test
+	public void testGetAllAttributes() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)